File: .trash/mapreduce_framework/wordcount_mapper.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <algorithm>
#include <cctype>

int main() {
    std::string line;
    
    while (std::getline(std::cin, line)) {
        // Convert to lowercase and extract words
        std::transform(line.begin(), line.end(), line.begin(), ::tolower);
        
        std::istringstream iss(line);
        std::string word;
        
        while (iss >> word) {
            // Remove punctuation
            word.erase(std::remove_if(word.begin(), word.end(), 
                      [](char c) { return std::ispunct(c); }), word.end());
            
            if (!word.empty()) {
                std::cout << word << "\t1" << std::endl;
            }
        }
    }
    
    return 0;
}

File: .trash/mapreduce_framework/wordcount_mapper.py
#!/usr/bin/env python3
"""
Word Count Mapper - Reads lines from stdin, outputs word counts to stdout
Output format: word\t1
"""
import sys
import re

def main():
    for line in sys.stdin:
        # Remove punctuation and convert to lowercase
        line = line.strip().lower()
        words = re.findall(r'\b\w+\b', line)
        
        # Output each word with count 1
        for word in words:
            if word:  # Skip empty words
                print(f"{word}\t1")

if __name__ == "__main__":
    main()
